src/yarp.c

```diff
@@ -1016,6 +1016,73 @@ yp_block_argument_node_create(yp_parser_t *parser, const yp_token_t *operator, y
     return node;
 }
 
+static void
+YP_ATTRIBUTE_UNUSED yp_scope_node_create(yp_parameters_node_t *parameters, yp_node_t *body, yp_constant_id_list_t locals, const char *start, const char *end, yp_scope_node_t *dest)
+{
+    *dest = (yp_scope_node_t) {
+         {
+             .type = YP_NODE_SCOPE_NODE,
+             .location = { .start = start, .end = end },
+         },
+        .parameters = parameters,
+        .body = body,
+        .locals = locals,
+    };
+    return;
+}
+
+// TODO: Implement this for all other nodes which can have a scope
+void
+yp_scope_node_init(yp_node_t *node, yp_scope_node_t *dest) {
+    yp_parameters_node_t *parameters = NULL;
+    yp_node_t *body;
+    yp_constant_id_list_t locals;
+    const char *start = node->location.start;
+    const char *end = node->location.end;
+
+    switch (node->type) {
+      case YP_NODE_BLOCK_NODE: {
+          yp_block_node_t *block_node = (yp_block_node_t *) node;
+          body = (yp_node_t *)block_node->body;
```

Comment:
These casts shouldn't be necessary here because they should already be `yp_node_t`